libs/sdk-core/src/breez_services.rs

```diff
@@ -917,13 +917,18 @@ impl BreezServices {
         Ok(())
     }
 
-    /// Connects to the selected LSP peer. If none is selected, this selects the first one from [`list_lsps`] and persists the selection.
+    /// Connects to the selected LSP peer.
+    /// This validates if the selected LSP is still in [`list_lsps`].
+    /// If not or no LSP is selected, it selects the first LSP in [`list_lsps`].
     async fn connect_lsp_peer(&self, node_pubkey: String) -> SdkResult<()> {
-        // Sets the LSP id, if not already set
-        if self.persister.get_lsp_id()?.is_none() {
-            if let Some(lsp) = self.lsp_api.list_lsps(node_pubkey).await?.first().cloned() {
-                self.persister.set_lsp_id(lsp.id)?;
-            }
+        let lsps = self.lsp_api.list_lsps(node_pubkey).await?;
+        if let Some(lsp) = self
```

Comment:
What if this is `None`, but we do have an lsp persisted already. Will that cause issues?

Reply:
It would mean `list_lsps` is empty. We would be back to `receive_payment` failing, which would also fail if the persisted lsp is not set.